assets/js/modules/tagmanager/components/setup/SetupForm.js

```diff
@@ -16,61 +16,123 @@
+import { STORE_NAME, FORM_SETUP, EDIT_SCOPE, SETUP_MODE_WITH_ANALYTICS } from '../../datastore/constants';
+import { STORE_NAME as MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
 import { STORE_NAME as CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
 import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { STORE_NAME as CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
 import { isPermissionScopeError } from '../../../../util/errors';
 import {
 	AccountSelect,
 	AMPContainerSelect,
-	FormInstructions,
 	WebContainerSelect,
 } from '../common';
-import StoreErrorNotice from '../../../../components/StoreErrorNotice';
+import Button from '../../../../components/button';
+import Link from '../../../../components/link';
+import SetupErrorNotice from './SetupErrorNotice';
+import SetupFormInstructions from './SetupFormInstructions';
 const { useSelect, useDispatch } = Data;
 
 export default function SetupForm( { finishSetup } ) {
 	const canSubmitChanges = useSelect( ( select ) => select( STORE_NAME ).canSubmitChanges() );
+	const singleAnalyticsPropertyID = useSelect( ( select ) => select( STORE_NAME ).getSingleAnalyticsPropertyID() );
+	const analyticsModuleActive = useSelect( ( select ) => select( CORE_MODULES ).isModuleActive( 'analytics' ) );
 	const hasEditScope = useSelect( ( select ) => select( CORE_USER ).hasScope( EDIT_SCOPE ) );
-	const autoSubmit = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'autoSubmit' ) );
+	const analyticsModuleReauthURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getAdminReauthURL() );
+	// Only select the initial autosubmit + submitMode once from form state which will already be set if a snapshot was restored.
+	const initialAutoSubmit = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'autoSubmit' ), [] );
+	const initialSubmitMode = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'submitMode' ), [] );
 
 	const { setValues } = useDispatch( CORE_FORMS );
+	const { activateModule } = useDispatch( CORE_MODULES );
 	const { submitChanges } = useDispatch( STORE_NAME );
-	const submitForm = useCallback( async ( event ) => {
-		event.preventDefault();
-		const { error } = await submitChanges();
-		if ( isPermissionScopeError( error ) ) {
-			setValues( FORM_SETUP, { autoSubmit: true } );
-		}
-		if ( ! error ) {
+	const dispatchAnalytics = useDispatch( MODULES_ANALYTICS );
+	const submitForm = useCallback( async ( { submitMode } = {} ) => {
+		const throwOnError = async ( func ) => {
+			const { error } = await func() || {};
+			if ( error ) {
+				throw error;
+			}
+		};
+		// We'll use form state to persist the chosen submit choice
+		// in order to preserve support for auto-submit.
+		// Set `inProgress` optimistically to avoid flashes of progress bar and content.
+		setValues( FORM_SETUP, { submitMode, inProgress: true } );
```

Comment:
@felixarntz I used `inProgress` here rather than `isNavigating` which we had previously as it is more accurate to what it represents. Even if we used something other than `ProgressBar`, this state would still be relevant.